### PR TITLE
Update Wallet Tab Layout

### LINKED ISSUE #750: Relayout Activity Screen
https://www.figma.com/file/x07XtWRI573FfO8duPMIEz/3.4.0?node-id=1%3A121

3 4 0 2021-07-01 at 11 14 02 AM

### LINKED ISSUE #751: Redesign No Activity Screen
https://www.figma.com/file/x07XtWRI573FfO8duPMIEz/3.4.0?node-id=6%3A3

3 4 0 2021-07-01 at 11 15 42 AM

### PR DESCRIPTION
**Pull request by matthewcarlreetz**
Update HeliumSelect to scroll selected item into view.
Fix Activity Filter Unmounting.
Move wallet chart to activity card.
Handle copy wallet address
Handle share. Cleanup unused components.
Remove WalletViewContainer.
Animate transitions.
Move some wallet view concerns from screen component.
Fix some filter transitions.
Fix wrong height calculation on some android devices.
Create empty wallet view.

Closes #750
Closes #751

### CODE REVIEW

File: src/components/BarChart/WalletChart.tsx [hunk 1]
```diff
@@ -100,16 +100,45 @@ const WalletChart = ({ height }: Props) => {
...
+  )
+
+  const handleHeaderLayout = useCallback(
+    (e: LayoutChangeEvent) => setHeaderHeight(e.nativeEvent.layout.height),
+    [],
+  )
+
   const { greenBright, blueBright } = useColors()
 
+  // TODO: Create a skeleton loader
```
Comment (matthewcarlreetz) [inline src/components/BarChart/WalletChart.tsx:131]: Will handle this as part of the activity listview redesign

File: src/components/BarChart/WalletChart.tsx [hunk 2]
```diff
@@ -50,6 +59,9 @@ const WalletChart = ({ height }: Props) => {
   }, [focusedData?.down, focusedData?.up, hntToDisplayVal])
   const [dataRange, setDataRange] = useState('')
 
+  // TODO: Figure out why re-loading the same chart takes so long
```
Comment (matthewcarlreetz) [inline src/components/BarChart/WalletChart.tsx:62]: Will handle this as part of the activity listview redesign

File: src/features/wallet/root/WalletScreen.tsx [hunk 3]
```diff
@@ -162,16 +162,24 @@ const WalletScreen = () => {
...
+        {(activityViewState === 'activity' ||
+          activityViewState === 'undetermined') && (
+          <WalletView
+            activityViewState={activityViewState}
+            showSkeleton={showSkeleton}
+            txns={transactionData}
+            pendingTxns={pendingTxns}
+          />
+        )}
+        {/* TODO: Handle state where user has no activity and no balance */}
```
Comment (matthewcarlreetz) [inline src/features/wallet/root/WalletScreen.tsx:175]: Will handle this in #751

File: src/features/wallet/root/WalletEmpty.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
+import React, { memo } from 'react'
+import { useTranslation } from 'react-i18next'
+import Box from '../../../components/Box'
+import Text from '../../../components/Text'
+
+const WalletEmpty = () => {
+  const { t } = useTranslation()
+
+  return (
+    <>
```
Comment (tyler-whitman) [inline src/features/wallet/root/WalletEmpty.tsx:10]: Fragment shouldn't be needed

File: src/features/wallet/root/WalletHeaderCondensed.tsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+      {...boxProps}
+    >
+      <Text
+        flex={1}
+        variant="h1"
+        color="white"
+        fontSize={22}
+        maxFontSizeMultiplier={1.2}
+      >
+        {`${balance.integerPart} HNT`}
```
Comment (tyler-whitman) [inline src/features/wallet/root/WalletHeaderCondensed.tsx:44]: When you use currency, the "HNT" shouldn't be appended. Also do we want to support tapping this to toggle between HNT and currency?

File: src/features/wallet/root/WalletView.tsx [hunk 6]
```diff
@@ -58,50 +127,129 @@ const WalletView = ({
...
+              balanceInfo={balanceInfo}
+              account={account}
+              accountLoading={
+                fetchAccountState === 'idle' || fetchAccountState === 'rejected'
+              }
+              toggleConvertHntToCurrency={toggleConvertHntToCurrency}
+            />
+          </Box>
+        )}
+        {activityViewState === 'no_activity' && <WalletEmpty />}
```
Comment (tyler-whitman) [inline src/features/wallet/root/WalletView.tsx:227]: I see this while the activity is initially loading, I think before there was a spinner. Was this an intended change?

**Review by tyler-whitman — APPROVED** [review]
Looks good, just a couple minor comments

File: src/features/wallet/root/WalletView.tsx (hunk 6 above)
Comment (matthewcarlreetz) [reply src/features/wallet/root/WalletView.tsx:227]: Yeah, that animation is intended. I guess we'll see what people think.

File: src/features/wallet/root/WalletHeaderCondensed.tsx (hunk 5 above)
Comment (matthewcarlreetz) [reply src/features/wallet/root/WalletHeaderCondensed.tsx:44]: Definitely to the toggle, good call.

File: src/features/wallet/root/WalletView.tsx (hunk 6 above)
Comment (tyler-whitman) [reply src/features/wallet/root/WalletView.tsx:227]: Ok cool, looks good then

File: src/features/wallet/root/WalletHeaderCondensed.tsx (hunk 5 above)
Comment (matthewcarlreetz) [reply src/features/wallet/root/WalletHeaderCondensed.tsx:44]: And you're right about the formatting - I forgot about that 🤦

**Pull request merged**